src/averageAges.js

```diff
@@ -37,7 +48,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let ages;
+  const womens = people.filter(person => person.sex === 'f');
+
+  if (withChildren !== undefined) {
+    const peopleMothers = people.map(person => person.mother);
+
+    ages = [...womens]
+      .filter(women => peopleMothers.includes(women.name))
+      .map(women => women.died - women.born);
+  } else {
+    ages = [...womens].map(women => women.died - women.born);
+  }
```

Comment:
Please try to silmlify it, just filter people whicjh you need and make withChildren checking inside a filter callback.
Avoid if else.